java/java.editor/src/org/netbeans/modules/java/editor/fold/JavaElementFoldManager.java

```diff
@@ -273,6 +277,11 @@ public FoldInfo createInnerClassFold(int start, int end) {
                     return FoldInfo.range(start, end, INNERCLASS_TYPE);
                 }
 
+                @Override
+                public FoldInfo createMethodsFold(int start, int end) {
```

Comment:
I know that it may be awkward - given the 'createCodeBlockFold` really does not describe 'method-level' folds well (although it has been its purpose) -- but since there are known implementation dependencies, maybe the `Creator`'s method could rremain as it is - and the **real** code blocks could get a new name ?

cc @dbalek -- what's your opinion ? Impl dependencies usually do not deserve any compatibility attention (especially if this PR refactors all known ones in the codebase), but still - this support incl. the misnamed code block (both constant and the Creator method) is used for long time by plugins.

Reply:
I already thought about this. I mean a method is also a code block as you said. But yeah at the end I'm fine with this and not to add for really each case a method to have also an option for if, for for, etc.